ingest-app/src/cmr/ingest/services/bulk_update_service.clj

```diff
@@ -114,13 +115,32 @@
     (when (seq concepts)
       (distinct (map :concept-id concepts)))))
 
+(defn- get-validated-concept-ids
+  "Returns the concept-ids that are valid. Throws exception if there exists invalid
+   concept-ids."
+  [context concept-ids provider-id]
+  (if (= ["ALL"] (map string/upper-case concept-ids)) 
+    (get-provider-concept-ids context provider-id)
+    (let [err-msgs 
+           (for [id concept-ids
+                 :let [err-msg (concepts/concept-id-validation id)
+                       err-msg (if err-msg
+                                 err-msg
+                                 (when-not (string/starts-with? id "C")
+                                   [(str "Collection concept-id " id " does not start with C")]))]
+                 :when err-msg]
+             err-msg)]
+      (if (seq err-msgs)
+        (errors/throw-service-errors
+          :bad-request
+          [(string/join ", " err-msgs)])
+        concept-ids))))
+
 (defn- get-concept-ids
   "Get the concept-ids from either the concept-ids passed in or 
    from the provider. If both are empty, throws exception." 
   [context concept-ids provider-id]
-  (if-let [concept-ids (if (seq concept-ids)
-                         concept-ids
-                         (get-provider-concept-ids context provider-id))]
+  (if-let [concept-ids (get-validated-concept-ids context concept-ids provider-id)]
```

Comment:
I think we should separate the validation from getting concept ids function. This way, we would have a definitive error message as to if there is no collection for the given provider or the provided concept ids list is empty.

Reply:
Done.